### ISSUE TITLE
Suite of suites testng.xml parallel execution

### ISSUE THREAD

**Issue by petermartin2002** [testng-team/testng#1686]
### TestNG Version
```xml
            <dependency>
                <groupId>org.testng</groupId>
                <artifactId>testng</artifactId>
                <version>6.9.10</version>
                <scope>test</scope>
            </dependency>
```

### Expected behavior
When running a suite of suites via testing.xml I expect each class within a suite to run in parallel.

### Actual behaviour
Each suite runs in parallel but classes within the suite run sequentially:
suitea testcase 1 and suite b testcase2 run in parallel 
suitea testcase3 runs after testcase1 completes

### Is the issue reproductible on runner?
 IntelliJ

### Test case sample
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="3" name="Suite" parallel="tests">
    <suite-files>
        <suite-file path="suitea.xml" />
        <suite-file path="suiteb.xml" />
    </suite-files>
</suite> <!-- Suite -->
```

------suitea.xml--------------
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="3" name="Suite" parallel="tests">
    <test name="FirstTest" thread-count="5" parallel="methods" >
        <parameter name="browser" value="chrome"/>
        <parameter name = "title" value="SS1 Verify the login section"/>
        <classes>
            <class name="testsuite.TestCase1"/>
        </classes>
    </test> <!-- First Test -->
    <test name="SecondTest" thread-count="5" parallel="methods" >
        <parameter name="browser" value="chrome"/>
        <parameter name = "title" value="SS2 Second Login Test"/>
        <classes>
            <class name="testsuite.TestCase3"/>
        </classes>
    </test> <!-- Second Test -->
</suite> <!-- Suite -->
```
------suiteb.xml--------------
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="3" name="Suite" parallel="tests">

    <test name="SecondTest" thread-count="5" parallel="methods" >
        <parameter name="browser" value="chrome"/>
        <parameter name = "title" value="SR1 Verify item in mobile list page can be sorted by name"/>
        <classes>
            <class name="testsuite.TestCase2"/>
        </classes>
    </test> <!-- First Test -->
</suite> <!-- Suite -->
```

**Comment by krmahadevan** [comment]
@petermartin2002 - Can you please retry using the latest released version of TestNG `6.14.2` and let us know if the issue exists ? `6.9.10` is a very old version.

**Comment by petermartin2002** [comment]
Upgraded and still the same issue

**Comment by krmahadevan** [comment]
@petermartin2002 - Can you please help share a sample that can be used to recreate this issue ?

**Comment by petermartin2002** [comment]
[testsuite.zip](https://github.com/cbeust/testng/files/1702337/testsuite.zip)
[framework.zip](https://github.com/cbeust/testng/files/1702338/framework.zip)

Test cases, pom.xml, and testng, and suite xml attached.

It is really straightforward to recreate from that.

**Comment by petermartin2002** [comment]
Any advice here folks?

When I run the suites on their own they run in parallel but the suite of suites runs each suite and its subsequent tests serially.  Quite frustrating

**Comment by krmahadevan** [comment]
@petermartin2002 - The attached project contains references to some classes that aren't included. So its not possible to reproduce the issue.

I would request you to please attach a simple test project (without any page objects or selenium dependency), that can be used to reproduce the issue.

**Comment by glenn04** [comment]
<img width="445" alt="screen shot 2018-04-20 at 10 28 34 am" src="https://user-images.githubusercontent.com/28329412/39021597-97d6874c-4485-11e8-9d7d-31afd1de7393.png">

**Comment by krmahadevan** [comment]
@glenn04 - Can you please help create a simple standalone project (which uses nothing outside of core java) that we can use to reproduce the problem ?

**Comment by albanlorillard** [comment]
Hello,
@glenn04  and @petermartin2002  Have you fix your problem ? Because I have exactly the same issue... Please here is a simple project without other dependencies. Ready to use !
[issues1686.zip](https://github.com/cbeust/testng/files/2093484/issues1686.zip)
To execute : 
`mvn clean verify -DsuiteXmlFile=myTestng.xml`

Thank you :)

**Comment by glenn04** [comment]
@albanlorillard not yet bro, no answer yet from @krmahadevan ?

**Comment by krmahadevan** [comment]
@glenn04 - I will get to this as soon as possible. A bit swamped by stuff.

@albanlorillard - Thanks for providing a reproducible test. I will try using that to see what is going on.

**Comment by DhavalAtri** [comment]
Hello @krmahadevan ,
Please consider below sample code.

public class LoginTest {
	@Test (priority=0)
	public void loginTest(){
		System.out.println("User Login");
	}
        @Test (priority=1)
	public void registerUser(){
		System.out.println("Register User ");
	}
	@Test (priority=2)
	public void forgotPass(){
		System.out.println("Forgot Password");
	}
}

public class addProduct {
	@Test (priority=0)
	public void ProductAdd() {
		System.out.println("Product Added ");
	}
	@Test(priority=1)
	public void editProduct(){
		System.out.println("Product Edited");
	}
	@Test(priority=2)
	public void deleteProduct(){
		System.out.println("Delete Product");
	}
}

**Testng.xml**

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="Suite">
  <test name="Test">
    <classes>
      <class name="test.demo.org.addProduct"/>
      <class name="test.demo.org.LoginTest"/>
    </classes>
  </test> 
</suite> 

**pom.xml**
	<dependencies>
		<dependency>
			<groupId>org.testng</groupId>
			<artifactId>testng</artifactId>
			<version>6.14.3</version>
			<scope>test</scope>
		</dependency>
	</dependencies>

Also my all test run in parallel.
**Out Put :**
Product Added 
User Login
Product Edited
Register User 
Delete Product
Forgot Password

===============================================
Suite
Total tests run: 6, Failures: 0, Skips: 0
===============================================

**Comment by VishnuGupta16** [comment]
Did any one found any solution ?

**Comment by DhavalAtri** [comment]
if you make yours testng.xml as shown below then your test will execute in proper manner.
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="Suite">
  <test name="Test">
    <classes>
      <class name="com.testNGXML.org.LoginTest"/>
    </classes>
  </test> 
  <test name="Test">
    <classes>
      <class name="com.testNGXML.org.RegisterUserTest"/>
    </classes>
  </test> 
</suite> 


Share your experience or thoughts 
@VishnuGupta16

**Comment by tarasmytlovych** [comment]
@DhavalAtri But it is not suite of suites
The question was about suite of suites parallel run

**Comment by tarasmytlovych** [comment]
Any updates on this?
I am using cucumber testng and have the same problem with suite of suites (that listed there suited runs one after another, not in parallel)

**Comment by krmahadevan** [comment]
@tarasmytlovych - No I am yet to get to this issue. Is this a problem when using TestNG `7.0.0-beta3` (latest released version as of today) ?

**Comment by krmahadevan** [comment]
I have tried reproducing this issue with TestNG `7.0.0-beta3` and I can't seem to recreate it

@tarasmytlovych - Unless you have a reproducible test that can still reproduce the problem, I may have to close this issue.

Here's what I have (Its the same example that @petermartin2002 had shared when the bug was created. I just cleaned it up and added thread id printing)

```java
package com.rationaleemotions.github.issue1686;

import org.testng.ITestResult;
import org.testng.Reporter;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class TestCase1 {
  @BeforeMethod
  public void beforeMethod() {
    System.err.println(getPrefix() + getClass().getName() + " triggered beforeMethod() on #" + Thread.currentThread().getId());
  }

  @Test(priority = 1, description = "SS1 Verify the login section")
  public void login() {
    System.err.println(getPrefix() + getClass().getName() + " triggered login() on #" + Thread.currentThread().getId());
  }

  @AfterMethod
  @Parameters("title")
  public void validate(String title) {
    System.err.println(getPrefix() + getClass().getName() + " triggered validate() with parameter [" + title + "] on #" + Thread.currentThread().getId());
  }

  private static String getPrefix() {
    ITestResult result = Reporter.getCurrentTestResult();
    String suiteName = result.getTestContext().getSuite().getName();
    String testTagName = result.getTestContext().getName();
    return "[Suite :<" + suiteName + ">, Test: <" + testTagName + "> ";
  }

}
```

```java
package com.rationaleemotions.github.issue1686;

import org.testng.ITestResult;
import org.testng.Reporter;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class TestCase2 {
  @BeforeMethod
  public void beforeMethod() {
    System.err.println(getPrefix() + getClass().getName() + " triggered beforeMethod() on #" + Thread.currentThread().getId());
  }

  @Test(priority = 1, description = "SR1 Verify item in mobile list page can be sorted by name")
  public void SR1() {
    System.err.println(getPrefix() + getClass().getName() + " triggered SR1() on #" + Thread.currentThread().getId());
  }

  @AfterMethod
  public void validate(){
    System.err.println(getPrefix() + getClass().getName() + " triggered validate() on #" + Thread.currentThread().getId());
  }

  private static String getPrefix() {
    ITestResult result = Reporter.getCurrentTestResult();
    String suiteName = result.getTestContext().getSuite().getName();
    String testTagName = result.getTestContext().getName();
    return "[Suite :<" + suiteName + ">, Test: <" + testTagName + "> ";
  }

}
```

```java
package com.rationaleemotions.github.issue1686;

import org.testng.ITestResult;
import org.testng.Reporter;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class TestCase3 {
  @BeforeMethod
  public void beforeMethod() {
    System.err.println(
        getPrefix()
            + getClass().getName()
            + " triggered beforeMethod() on #"
            + Thread.currentThread().getId());
  }

  @Test(priority = 1, description = "SS1 Verify the login section")
  public void login() {
    System.err.println(
        getPrefix()
            + getClass().getName()
            + " triggered login() on #"
            + Thread.currentThread().getId());
  }

  @AfterMethod
  @Parameters("title")
  public void validate(String title) {
    System.err.println(
        getPrefix()
            + getClass().getName()
            + " triggered validate() with parameter ["
            + title
            + "] on #"
            + Thread.currentThread().getId());
  }

  private static String getPrefix() {
    ITestResult result = Reporter.getCurrentTestResult();
    String suiteName = result.getTestContext().getSuite().getName();
    String testTagName = result.getTestContext().getName();
    return "[Suite :<" + suiteName + ">, Test: <" + testTagName + "> ";
  }
}
```

Suites look like below

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="3" name="Suite" parallel="tests">
  <test name="FirstTest" thread-count="5" parallel="methods">
    <parameter name="browser" value="chrome"/>
    <parameter name="title" value="SS1 Verify the login section"/>
    <classes>
      <class name="com.rationaleemotions.github.issue1686.TestCase1"/>
    </classes>
  </test> <!-- First Test -->
  <test name="SecondTest" thread-count="5" parallel="methods">
    <parameter name="browser" value="chrome"/>
    <parameter name="title" value="SS2 Second Login Test"/>
    <classes>
      <class name="com.rationaleemotions.github.issue1686.TestCase3"/>
    </classes>
  </test> <!-- Second Test -->
</suite> <!-- Suite -->
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="3" name="Suite" parallel="tests">

  <test name="SecondTest" thread-count="5" parallel="methods" >
    <parameter name="browser" value="chrome"/>
    <parameter name = "title" value="SR1 Verify item in mobile list page can be sorted by name"/>
    <classes>
      <class name="com.rationaleemotions.github.issue1686.TestCase3"/>
    </classes>
  </test> <!-- First Test -->
</suite> <!-- Suite -->
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="3" name="Suite" parallel="tests" verbose="2">
  <suite-files>
    <suite-file path="./suitea.xml" />
    <suite-file path="./suiteb.xml" />
  </suite-files>
</suite> <!-- Suite -->
```

Execution output

```
...
... TestNG 7.0.0-beta3 by Cédric Beust (cedric@beust.com)
...
[Suite :<Suite (0)>, Test: <SecondTest> com.rationaleemotions.github.issue1686.TestCase3 triggered beforeMethod() on #13
[Suite :<Suite (0)>, Test: <FirstTest> com.rationaleemotions.github.issue1686.TestCase1 triggered beforeMethod() on #14
[Suite :<Suite (0)>, Test: <FirstTest> com.rationaleemotions.github.issue1686.TestCase1 triggered login() on #14
[Suite :<Suite (0)>, Test: <SecondTest> com.rationaleemotions.github.issue1686.TestCase3 triggered login() on #13
[Suite :<Suite (0)>, Test: <FirstTest> com.rationaleemotions.github.issue1686.TestCase1 triggered validate() with parameter [SS1 Verify the login section] on #14
[Suite :<Suite (0)>, Test: <SecondTest> com.rationaleemotions.github.issue1686.TestCase3 triggered validate() with parameter [SS2 Second Login Test] on #13

===============================================
Suite (0)
Total tests run: 2, Passes: 2, Failures: 0, Skips: 0
===============================================
[Suite :<Suite (1)>, Test: <SecondTest> com.rationaleemotions.github.issue1686.TestCase3 triggered beforeMethod() on #16
[Suite :<Suite (1)>, Test: <SecondTest> com.rationaleemotions.github.issue1686.TestCase3 triggered login() on #16
[Suite :<Suite (1)>, Test: <SecondTest> com.rationaleemotions.github.issue1686.TestCase3 triggered validate() with parameter [SR1 Verify item in mobile list page can be sorted by name] on #16

===============================================
Suite (1)
Total tests run: 1, Passes: 1, Failures: 0, Skips: 0
===============================================

===============================================
Suite
Total tests run: 3, Passes: 3, Failures: 0, Skips: 0
===============================================

```

**Comment by agoldenberg02** [comment]
Hi All:

Have anybody tried a suite of xml suites syntax with something :
<suite thread-count="3" name="Suite" parallel="xml">  ?
Have an xml suite os xml suites and need to find a proper syntax.
Thanks in advance for the advice!

**Comment by krmahadevan** [comment]
@agoldenberg02 

> `<suite thread-count="3" name="Suite" parallel="xml"> ?`

There is nothing like that. The only allowed values are 

1. methods
2. classes
3. tests
4. instances

**Comment by krmahadevan** [comment]
Haven't seen any response for my earlier question on this and since I am also not able to reproduce the problem. Closing this. If this is still a problem, need a reproducible sample using the latest released version of TestNG (`7.0.0-beta7` as of today)

**Comment by tngii** [comment]
@krmahadevan as far as i see, in your example **suites** are executed sequentially instead of parallel (first - suitea.xml, then suiteb.xml), but the main issue is how to run suitea and suiteb parallel

**Comment by goyal-harsh** [comment]
> @krmahadevan as far as i see, in your example **suites** are executed sequentially instead of parallel (first - suitea.xml, then suiteb.xml), but the main issue is how to run suitea and suiteb parallel

Hey , We can use the Maven Surefire Plugin to Run the suites in parallel by adding a property called <suitepoolthreadcount> to the Plugin Configuration.

**Comment by prvn82** [comment]
Even I am waiting for same issue's solution

**Comment by ZathrasOne** [comment]
This is exactly what I am observing as well, even with version 7.1.0. The suite-file's in a suite-files list are executed in sequence, instead of in parallel.

**Comment by krmahadevan** [comment]
Here's the suite of suite file

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="github-1686-master-suite" verbose="2">
  <suite-files>
    <suite-file path="slave-1.xml"/>
    <suite-file path="slave-2.xml"/>
  </suite-files>
</suite>
```

And here's how the surefire plugin looks like

```xml
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>3.0.0-M1</version>
        <executions>
          <execution>
            <phase>test</phase>
          </execution>
        </executions>
        <configuration>
          <suiteXmlFiles>
            <suiteXmlFile>src/test/resources/master.xml</suiteXmlFile>
          </suiteXmlFiles>
          <properties>
            <property>
              <name>suitethreadpoolsize</name>
              <value>${threads}</value>
            </property>
          </properties>
        </configuration>
      </plugin>
```

Execution output

```bash
[INFO] Running TestSuite
...
... TestNG 7.1.0 by Cédric Beust (cedric@beust.com)
...

Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-11>, MethodName: <com.rationaleemotions.TestCase1.one()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-21>, MethodName: <com.rationaleemotions.TestCase1.one()> running on Thread : 14
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-21>, MethodName: <com.rationaleemotions.TestCase1.two()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-11>, MethodName: <com.rationaleemotions.TestCase1.two()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-21>, MethodName: <com.rationaleemotions.TestCase2.one()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-11>, MethodName: <com.rationaleemotions.TestCase2.one()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-21>, MethodName: <com.rationaleemotions.TestCase2.two()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-11>, MethodName: <com.rationaleemotions.TestCase2.two()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-22>, MethodName: <com.rationaleemotions.TestCase1.one()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-12>, MethodName: <com.rationaleemotions.TestCase1.one()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-22>, MethodName: <com.rationaleemotions.TestCase1.two()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-12>, MethodName: <com.rationaleemotions.TestCase1.two()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-22>, MethodName: <com.rationaleemotions.TestCase2.one()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-12>, MethodName: <com.rationaleemotions.TestCase2.one()> running on Thread : 13
Suite: <github-1686-slave-suite-2>, TestName: <github-1686-test-22>, MethodName: <com.rationaleemotions.TestCase2.two()> running on Thread : 14
Suite: <github-1686-slave-suite-1>, TestName: <github-1686-test-12>, MethodName: <com.rationaleemotions.TestCase2.two()> running on Thread : 13
[INFO] Tests run: 16, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.515 s - in TestSuite
```

The below attached project should demonstrate the parallelism in action.
[github-1686.zip](https://github.com/cbeust/testng/files/4672989/github-1686.zip)

If this is still a problem, please create a new issue and attach a sample project that can be used to reproduce the problem.

**Comment by fh-saikumar** [comment]
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="API_PartnerAPI_Regression">
    <suite-files>
        <suite-file path="API_PAPIUberEats_Regression.xml"></suite-file>
        <suite-file path="API_JUSTEAT_Regression.xml"></suite-file>
        <suite-file path="API_Deliveroo_Regression.xml"></suite-file>
        <suite-file path="API_Zuppler_Regression.xml"></suite-file>
    </suite-files>
</suite>

**Comment by fh-saikumar** [comment]
how can i execute parallel for below  suite files????

 <!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="API_PartnerAPI_Regression">
    <suite-files>
        <suite-file path="API_PAPIUberEats_Regression.xml"></suite-file>
        <suite-file path="API_JUSTEAT_Regression.xml"></suite-file>
        <suite-file path="API_Deliveroo_Regression.xml"></suite-file>
        <suite-file path="API_Zuppler_Regression.xml"></suite-file>
    </suite-files>
</suite>

**Comment by krmahadevan** [comment]
@fh-saikumar Please use the [TestNG google groups](https://groups.google.com/g/testng-users) to get help on TestNG. 
And when you post on the google forum, please ensure that you share more context so that it's easier to understand what is the issue. Dont forget to use the latest released version of TestNG (7.6.1 as of today)

**Comment by krmahadevan** [comment]
That's a suite of suites. I don't think you can execute them in parallel. Atleast not as far as I know of.

**Comment by silverenio** [comment]
So, the final answer, that it is impossible to run separated suites in parallel from master suite?
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="Automation Test Suite" parallel="tests" thread-count="10">
    <suite-files>
        <suite-file path="src/test/resources/suites/web.xml"/>
        <suite-file path="src/test/resources/suites/api.xml"/>
    </suite-files>
</suite>
```

![2024-03-05_13h51_27](https://github.com/testng-team/testng/assets/8516873/077bb1a0-0c50-45b9-a05b-1c058d4e6c44)

**Comment by krmahadevan** [comment]
@silverenio - Yeah that's the current behaviour as of today. Perhaps it can be changed. 

@juherr - WDYT ?

**Comment by krmahadevan** [comment]
@silverenio - I checked this once again and it looks like it can be done. Please check the below attached project.

when you run the tests using `mvn clean test` you should see an output like below


```bash
[INFO] --- surefire:3.2.5:test (default-test) @ issue_1686 ---
[INFO] Using auto detected provider org.apache.maven.surefire.testng.TestNGProvider
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running TestSuite
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
...
... TestNG 7.9.0 by Cédric Beust (cedric@beust.com)
...



[Thread #19] Suite -> <child_suite_2>, Test -> <child_test_2>, TestMethod -> <com.rationaleemotions.SampleTestCase.test1()>
[Thread #18] Suite -> <child_suite_1>, Test -> <child_test_1>, TestMethod -> <com.rationaleemotions.SampleTestCase.test1()>

[Thread #19] Suite -> <child_suite_2>, Test -> <child_test_2>, TestMethod -> <
com.rationaleemotions.SampleTestCase.test2()>
[Thread #18] Suite -> <child_suite_1>, Test -> <child_test_1>, TestMethod -> <com.rationaleemotions.SampleTestCase.test2()>
PASSED: com.rationaleemotions.SampleTestCase.test2
PASSED: com.rationaleemotions.SampleTestCase.test1
PASSED: com.rationaleemotions.SampleTestCase.test2
PASSED: com.rationaleemotions.SampleTestCase.test1

===============================================
    child_test_2
    Tests run: 2, Failures: 0, Skips: 0
===============================================


===============================================
    child_test_1
    Tests run: 2, Failures: 0, Skips: 0
===============================================

[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.608 s -- in TestSuite
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipp

ed: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.039 s
[INFO] Finished at: 2024-03-05T19:34:03+05:30
[INFO] ------------------------------------------------------------------------


```

[issue_1686.zip](https://github.com/testng-team/testng/files/14496370/issue_1686.zip)
